app/src/lib/dispatcher/app-shell.ts

```diff
@@ -1,15 +1,56 @@
 import { shell as electronShell } from 'electron'
 
+export interface IEditorInfo {
+  readonly name: string
+}
+
 export interface IAppShell {
   readonly moveItemToTrash: (path: string) => boolean
   readonly beep: () => void
   readonly openExternal: (path: string) => void
   readonly openItem: (path: string) => boolean
+  readonly getEditors: (path: string) => IEditorInfo[]
+}
+
+function getEditorList(path: string ): IEditorInfo[] {
+  const result = new Array<IEditorInfo>()
+  result.push({
+    name: 'External Editor',
+  })
+  if (__DARWIN__) {
+    const osx = require('shell-osx')
+    if (osx.isAtomInstalled()) {
+    result.push({
+      name: 'Atom',
+    })
+    }
```

Comment:
OSX part of the code is non-existent right now.  In fact it doesn't even check if Atom is installed
Plan to support VSCode.